src/components/Tooltip/Tooltip.tsx

```diff
@@ -1,24 +1,63 @@
-import React from 'react';
+import React, { useRef, useEffect, useState } from 'react';
 import { TooltipProps } from './types';
-import TooltipContainer from './Tooltip.styles';
+import {
+    StyledHoverDiv,
+    StyledTooltipParentDiv,
+    StyledPopoverArrowDiv,
+    StyledTooltipTextDiv,
+} from './Tooltip.styles';
 
 const Tooltip: React.FC<TooltipProps> = ({
-    id = String(Date.now()),
     position = 'bottom',
-    text,
+    content,
     children,
     maxWidth,
 }: TooltipProps) => {
+    const parretnRef = useRef(null);
+    const popoverRef = useRef(null);
+    const [width, setWidth] = useState(0);
+    const [height, setHeight] = useState(0);
+    const [popverWidth, setPopoverWidth] = useState(0);
+    const [popoverHeight, setPopoverHeight] = useState(0);
+    useEffect(() => {
+        console.log(
+            'width',
+            parretnRef.current ? (parretnRef.current as any).offsetWidth : 0,
+        );
+        setWidth(
+            parretnRef.current ? (parretnRef.current as any).offsetWidth : 0,
+        );
+        setHeight(
+            parretnRef.current ? (parretnRef.current as any).offsetHeight : 0,
+        );
+        setPopoverWidth(
+            popoverRef.current ? (popoverRef.current as any).offsetWidth : 0,
+        );
+        setPopoverHeight(
+            popoverRef.current ? (popoverRef.current as any).offsetHeight : 0,
+        );
+    }, [parretnRef.current, popoverRef.current]);
     return (
-        <TooltipContainer
-            id={id}
-            position={position}
-            data-tooltip={text}
+        <StyledTooltipParentDiv
+            ref={parretnRef}
             data-testid={'tooltip-container-test-id'}
-            maxWidth={maxWidth}
         >
-            <div data-testid={'tooltip-children-test-id'}>{children}</div>
-        </TooltipContainer>
+            <span>{children}</span>
```

Comment:
as we are allowing to pass in jsx elements I dont think its good allowing to pass in `div` inside `span`.

What is the actual difference between children and content?

Reply:
Children is for what is shown 
content is what's shown in the tooltip